File: client/src/app/core/models/conversation.ts
import { MessageResponse } from "./message";

export interface ConversationResponse {
    id: number;
    title: string;
    isGroup: boolean,
    unreadCount: number;
    isOnline?: boolean;
    participantIds: number[];
    participantNames: string[];
    photoUrl: string | null;
}

export interface ConversationDetails {
    id: number;
    title: string;
    messages: MessageResponse[];
    participants: ParticipantNames[];
    adminId: number;
}

export interface CreateConversationData {
    title: string;
    participantIds: number[];
}

export interface RemoveUserRequest {
    userId: number;
    conversationId: number;
}

export interface AddUsersRequest {
    userIds: number[];
    conversationId: number;
}

export interface ParticipantNames {
    username: string;
    userId: number;
    photoUrl: string;
}

export interface SearchConversationRequest{
    conversationId: number;
    filter: string;
}

File: server/Domain/Entities/Conversation.cs
namespace Domain.Entities
{
    public class Conversation
    {
        public int Id { get; set; }
        public string? Title { get; set; }
        public bool IsGroup { get; set; }
        public int LastMessageAddedId { get; set; }
        public int UnreadCount { get; set; }
        public List<Participation> Participants { get; set; } = new();
        public List<Message> Messages { get; set; } = new();
        public int AdminId { get; set; }
        public User? Admin { get; set; } = null;
    }
}
